src/ipc/ipc4/helper.c

```diff
@@ -499,6 +499,9 @@ static void construct_config(struct ipc4_copier_module_cfg *copier_cfg, uint32_t
 	} else if (fifo_size % 16 == 0) {
 		copier_cfg->base.audio_fmt.sampling_frequency = IPC4_FS_16000HZ;
 		frame_size = fifo_size / 16;
+	} else {
+		tr_err(&comp_tr, "unsupported fifo_size %d", fifo_size);
+		return -EINVAL;
```

Comment:
IPC4 doesn't use POSIX error codes

Reply:
@tmleman @marcinszkudlinski @mmaka1 @RanderWang which error code should be returned here ? If no one know we can fix later as this is a build breakage.

Reply:
First I thought it is not the final error, so use EINVAL. According to following advice by lyakh, we can use IPC4_ERROR_INVALID_PARAM

Reply:
Fixed now.